tests/Unit/TestingFramework.hpp

```diff
@@ -212,6 +212,60 @@ struct check_iterable_approx<
     }
   }
 };
+
+#define CHECK_MATRIX_APPROX(a, b)                                            \
+  do {                                                                       \
+    INFO(__FILE__ ":" + std::to_string(__LINE__) + ": " #a " == " #b);       \
+    check_matrix_approx<std::common_type_t<                                  \
+        std::decay_t<decltype(a)>, std::decay_t<decltype(b)>>>::apply(a, b); \
+  } while (false)
+
+#define CHECK_MATRIX_CUSTOM_APPROX(a, b, appx)                               \
+  do {                                                                       \
+    INFO(__FILE__ ":" + std::to_string(__LINE__) + ": " #a " == " #b);       \
+    check_matrix_approx<std::common_type_t<                                  \
+        std::decay_t<decltype(a)>, std::decay_t<decltype(b)>>>::apply(a, b,  \
+                                                                      appx); \
+  } while (false)
+
+template <typename M>
+struct check_matrix_approx {
+  // clang-tidy: non-const reference
+  static void apply(const M& a, const M& b,
+                    Approx& appx = approx) {  // NOLINT
+    // Can we adapt this so it works for both row-major and column-major
```

Comment:
remove or reword so that it is a statement about why it is done this way instead of as a question